coucharchive

```diff
@@ -282,6 +282,28 @@ def replicate_couchdb_server(source_url, target_url,
         pool.join()
 
 
+def bug_1418_create_missing_documents(source_db, target_db):
+    # Temporary function to be deleted when CouchDB team fixes the bug.
+    # It manually creates document on the target database that where
+    # previously deleted and re-created on the source.
+
+    missing_documents = [doc for doc in source_db if doc not in target_db]
+
+    for doc_id in missing_documents:
+        source_doc = source_db[doc_id]
+
+        # Not needed. Will be generated on target.
+        # Plus avoids conflicts.
+        del source_doc['_rev']
+
+        _, target_rev = target_db.save(source_doc)
+
+        if target_rev != source_doc.rev:
```

Comment:
One last thing I'm thinking about: in this case, it would be best to delete the extra bad doc we just created, in order to be able to re-run coucharchive on something clean:
```python
target_db.delete({'_id': source_doc, '_rev': target_rev})
```
(I'm not sure about the exact syntax)

Reply:
Thanks for pointing this out.

I also noticed a bug as `source_doc.rev` would not exists because of the previous `del source_doc['_rev']`.

PR updated with:
```diff
@@ -291,6 +291,7 @@ def bug_1418_create_missing_documents(source_db, target_db):
 
     for doc_id in missing_documents:
         source_doc = source_db[doc_id]
+        source_doc_rev = source_doc['_rev']
 
         # Not needed. Will be generated on target.
         # Plus avoids conflicts.
@@ -298,7 +299,10 @@ def bug_1418_create_missing_documents(source_db, target_db):
 
         _, target_rev = target_db.save(source_doc)
 
-        if target_rev != source_doc.rev:
+        if target_rev != source_doc_rev:
+            source_doc['_rev'] = target_rev
+            target_db.delete(source_doc)
+
             raise Exception('bug_1418_create_missing_documents: '
                             '%s: source and target documents have diverged'
                             % source_db.name)

```